### PR TITLE
Fix nullref + fix relative path for file archive

### PR DESCRIPTION
**Pull request by 304NotModified**
- Fix nullref  `<default-target-parameters` without  fileName
- Fix archive watch for relative path

fixes https://github.com/NLog/NLog/issues/1347

### CODE REVIEW

**Comment by 304NotModified** [comment]
The change in the filewatcher (fullpath) needs a review

**Comment by 304NotModified** [comment]
@bhaeussermann can you give your opinion on this small change?

File: src/NLog/Targets/FileTarget.cs [hunk 1]
```diff
@@ -605,6 +605,12 @@ private void RefreshFileArchive()
         {
             var nullEvent = LogEventInfo.CreateNullEvent();
             string fileNamePattern = GetArchiveFileNamePattern(GetCleanedFileName(nullEvent), nullEvent);
+            if (fileNamePattern == null)
+            {
+                InternalLogger.Debug("no RefreshFileArchive because fileName is NULL");
```
Comment (bhaeussermann) [inline src/NLog/Targets/FileTarget.cs]: Must this not be `Warn()` as well?

**Comment by bhaeussermann** [comment]
I had only one reservation regarding the `Debug()` call to `InternalLogger`. I'm happy with the rest ;)

**Comment by 304NotModified** [comment]
Thanks @bhaeussermann

**Pull request merged**